lib/runner/request-helpers-presend.js

```diff
@@ -17,6 +17,7 @@ module.exports = [
         // @todo - no need to sync vatiables when SDK starts supporting resolution from scope directly
         // @todo remove this hack of using empty variableList once it's fixed in sdk
         context.item = new sdk.Item(context.item.toObjectResolved({variables: new sdk.VariableList(null, [])}, [
+            context._variables.syncVariablesTo({}),
```

Comment:
@shamasis highlighting this change, `pm.variables.set` takes the highest precedence, higher than iteration data, the same as in `sandbox`